starter_code/src/movies.js

```diff
@@ -1,12 +1,92 @@
-/* eslint no-restricted-globals: 'off' */
-
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
 
+function orderByYear (arr) {
+	const newArray = [...arr];
+	const sortedYears = newArray.sort(function(a,b) {
+    if (a.year < b.year ) {
+			return -1;
+			}
+			else if (a.year > b.year) {
+				return 1;
+			}
+			else {
+				if (a.title < b.title) {
+					return -1;
+					}
+					else if (a.title > b.title) {
+						return 1;
+					}
+					else {
+						return 0;
+					} 
+      }			 
+		});
+	return sortedYears;
+}
+
+orderByYear(movies);
+
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(arr) {
+	
+	const sortDrama = arr.filter(function(e){
+		return e.genre.includes('Drama') && e.director === 'Steven Spielberg'; 
+		});	
+
+		return sortDrama.length;
+}
+
+howManyMovies(movies);
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
-// Iteration 4: All rates average - Get the average of all rates with 2 decimals
+function orderAlphabetically(arr) {
+	const newArray = [...arr];
+	const alphabetically = newArray.sort(function(a,b) {
+    if (a.title < b.title) {
+			return -1;
+			}
+			else if (a.title > b.title) {
+				return 1;
+			}
+			else {
+				return 0;
+			}	 
+		});
+	const first20 = newArray.slice(0, 19);
+
```

Comment:
The slice( ) method doesn't include the last given element, so here it would be newArray.slice(0, 20);